src/routes/boards.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const router = express.Router();
+const boardsData = require('../data/boards');
+
+/**
+ * Render base page that will show a user their boards
+ */
+router.get('/', async (req, res) => {
+    try {
+        //check if logged in
+        if(req.session.user) {
+            let userId = req.session.user.id;
+            if(userid) {
+                //read all of a user's boards
+                let boards = await boardsData.readAll(userId);
+                //render the boards page with their boards
+                res.render('boards', { title: "Boards", theBoards: boards, loggedIn: true });
+            }
+            else throw "Logged in user must have an ID";
```

Comment:
In general, it's better practice to use "throw new Error("");" as it displays the call stack. Change it if you want, otherwise, resolve this.

Reply:
I would like to be consistent and ALWAYS use throw new error(""); So please change it :)

Reply:
was unaware this was java, changed